samples/hello-terminal/terminal-workflow/src/main/java/com/squareup/sample/helloterminal/terminalworkflow/KeyStroke.kt

```diff
@@ -46,16 +49,18 @@ data class KeyStroke(
 @OptIn(ExperimentalCoroutinesApi::class)
 internal fun InputProvider.listenForKeyStrokesOn(
   scope: CoroutineScope
-): BroadcastChannel<KeyStroke> = scope.broadcast {
-  while (isActive) {
-    val keyStroke = readInput()
-    if (keyStroke.keyType === EOF) {
-      // EOF indicates the terminal input was closed, and we won't receive any more input, so
-      // close the channel instead of sending the raw event down.
-      return@broadcast
+): SharedFlow<KeyStroke> {
```

Comment:
Does this really need to be a `SharedFlow`? I was just mechanically following the conversion steps from the `BroadcastChannel` deprecation docs.